app/scripts/controllers/transactions.js

```diff
@@ -2,51 +2,62 @@ const EventEmitter = require('events')
 const extend = require('xtend')
 const ObservableStore = require('obs-store')
 const ethUtil = require('ethereumjs-util')
+const Transaction = require('ethereumjs-tx')
 const EthQuery = require('ethjs-query')
-const TxProviderUtil = require('../lib/tx-utils')
+const TransactionStateManger = require('../lib/tx-state-manager')
+const TxGasUtil = require('../lib/tx-gas-utils')
 const PendingTransactionTracker = require('../lib/pending-tx-tracker')
 const createId = require('../lib/random-id')
 const NonceTracker = require('../lib/nonce-tracker')
-const txStateHistoryHelper = require('../lib/tx-state-history-helper')
+
+/*
+  Transaction Controller is an aggregate of sub-controllers and trackers
+  composing them in a way to be exposed to the metamask controller
+    - txStateManager
+      responsible for the state of a transaction and
+      storing the transaction
+    - pendingTxTracker
+      watching blocks for transactions to be include
+      and emitting confirmed events
+    - txGasUtil
+      gas calculations and safety buffering
+    - nonceTracker
+      calculating nonces
+*/
 
 module.exports = class TransactionController extends EventEmitter {
   constructor (opts) {
     super()
-    this.store = new ObservableStore(extend({
-      transactions: [],
-    }, opts.initState))
-    this.memStore = new ObservableStore({})
     this.networkStore = opts.networkStore || new ObservableStore({})
     this.preferencesStore = opts.preferencesStore || new ObservableStore({})
-    this.txHistoryLimit = opts.txHistoryLimit
     this.provider = opts.provider
     this.blockTracker = opts.blockTracker
     this.signEthTx = opts.signTransaction
     this.ethStore = opts.ethStore
 
+    this.memStore = new ObservableStore({})
+    this.query = new EthQuery(this.provider)
```

Comment:
do we want to use `ethjs-rpc` instead of `ethjs-query` (no formatting, pending promisification of `ethjs-rpc`)

Reply:
update: `ethjs-rpc@0.1.7` is promised based :+1:

Reply:
could be into that.. will take a look